set-static-ip

```diff
@@ -11,13 +11,26 @@ if [ -z "$PROVISIONING_INTERFACE" ]; then
   # hard dependency of forcing users to specify the interface, we detect
   # which network interface has the IP's from the machine network.
   IP_ONLY=$(echo "$PROVISIONING_IP" | cut -d/ -f1)
-  PROVISIONING_INTERFACE=$(ip route get "$IP_ONLY" | grep -Po '(?<=(dev )).*(?= src| proto)')
+
+  # See if we already have the IP on an interface
+  PROVISIONING_INTERFACE=$(ip -j addr | jq -r -c ".[].addr_info[] | select(.local == \"$IP_ONLY\") | .label")
+
+  # If not, let's figure out what interface it should go on
+  if [ -z "$PROVISIONING_INTERFACE" ]; then
+    PROVISIONING_INTERFACE=$(ip -j route get "$IP_ONLY" | jq -r '.[] | select(.dev != "lo") | .dev')
+  fi
 else
   # Get rid of any DHCP addresses only if we have a dedicated
   # provisioning interface
   /usr/sbin/ip address flush dev "$PROVISIONING_INTERFACE"
 fi
 
+if [ -z "$PROVISIONING_INTERFACE" ]; then
```

Comment:
I think we're going to hit this error path every time the IP is already up?

e.g:

```
$ ip a s dev enp2s0
3: enp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:7b:d8:cc:c0:ab brd ff:ff:ff:ff:ff:ff
    inet 192.168.111.22/24 brd 192.168.111.255 scope global dynamic noprefixroute enp2s0
       valid_lft 3463sec preferred_lft 3463sec
    inet 192.168.111.5/24 scope global secondary enp2s0
       valid_lft forever preferred_lft forever
    inet6 fe80::27b:d8ff:fecc:c0ab/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever

$ ip -j route get "192.168.111.8" | jq -r '.[] | select(.dev != "lo") | .dev'
enp2s0

$ sudo ip addr add "192.168.111.8" dev enp2s0 valid_lft 300 preferred_lft 300

# Now we get an empty string... 
$ ip -j route get "192.168.111.8" | jq -r '.[] | select(.dev != "lo") | .dev'

$ ip -j route get "192.168.111.8"                             
[{"type":"local","dst":"192.168.111.8","dev":"lo","prefsrc":"192.168.111.8","flags":[],"uid":1000,"cache":["local"]}]

$ ip a s dev enp2s0
3: enp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:7b:d8:cc:c0:ab brd ff:ff:ff:ff:ff:ff
    inet 192.168.111.22/24 brd 192.168.111.255 scope global dynamic noprefixroute enp2s0
       valid_lft 3379sec preferred_lft 3379sec
    inet 192.168.111.8/32 scope global dynamic enp2s0
       valid_lft 278sec preferred_lft 278sec
    inet 192.168.111.5/24 scope global secondary enp2s0
       valid_lft forever preferred_lft forever
    inet6 fe80::27b:d8ff:fecc:c0ab/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever

$ ip a s dev lo    
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
```